packages/backend/src/telemetry/service.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  Counter,
+  DiagConsoleLogger,
+  DiagLogLevel,
+  MetricOptions,
+  diag,
+  metrics
+} from '@opentelemetry/api'
+import { OTLPMetricExporter } from '@opentelemetry/exporter-metrics-otlp-grpc'
+import { Resource } from '@opentelemetry/resources'
+import {
+  MeterProvider,
+  PeriodicExportingMetricReader
+} from '@opentelemetry/sdk-metrics'
+
+import { ConvertError, RatesService } from '../rates/service'
+import { ConvertOptions } from '../rates/util'
+import { BaseService } from '../shared/baseService'
+
+export interface TelemetryService {
+  getOrCreateMetric(name: string, options?: MetricOptions): Counter
+  getServiceName(): string | undefined
+  getBaseAssetCode(): string
+  getBaseScale(): number
+  convertAmount(
+    convertOptions: Omit<ConvertOptions, 'exchangeRate' | 'destinationAsset'>
+  ): Promise<bigint | ConvertError>
+}
+
+interface TelemetryServiceDependencies extends BaseService {
+  serviceName: string
+  collectorUrls: string[]
+  exportIntervalMillis?: number
+  aseRatesService: RatesService
+  fallbackRatesService: RatesService
+  baseAssetCode: string
+  baseScale: number
+}
+
+export function createTelemetryService(
+  deps: TelemetryServiceDependencies
+): TelemetryService {
+  return new TelemetryServiceImpl(deps)
+}
+
+class TelemetryServiceImpl implements TelemetryService {
+  private serviceName: string
+  private meterProvider?: MeterProvider
+  private fallbackRatesService: RatesService
+  private aseRatesService: RatesService
+
+  private counters = new Map()
+  constructor(private deps: TelemetryServiceDependencies) {
+    diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.DEBUG)
+    this.serviceName = deps.serviceName
+    this.fallbackRatesService = deps.fallbackRatesService
+    this.aseRatesService = deps.aseRatesService
+
+    if (
+      deps.collectorUrls &&
+      Array.isArray(deps.collectorUrls) &&
+      deps.collectorUrls.length === 0
+    ) {
+      deps.logger.info(
+        'No collector URLs specified, metrics will not be exported'
+      )
+      return
+    }
+
+    this.meterProvider = new MeterProvider({
+      resource: new Resource({ 'service.name': 'RAFIKI_NETWORK' })
```

Comment:
What's the service name here? Just something internal to the meter provider (I don't see it anywhere else)? Something that should match some configuration on somewhere AWS? I'm just wondering if this is the best way to set it - looks like a potential 'magic value'.

Reply:
The service name here is intended to be the same on all rafiki instrumented instances. it will serve as an aggregator property for metrics that are related to the "rafiki network / ILP Network" as a whole. By having it set on the sdk, all metrics will have their "job" property set as "RAFIKI_NETWORK".
You can think of it as the one property that identifies all the metrics with the same intent: Observability of the Rafiki/ILP Network.